content/blog/2021/09/15/runBenchmarkJDKComparison.sh.txt

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+# To prepare enviromnent
+########################
+# mkdir downloads
+# mkdir jdk
+# mkdir maven
+
+# To add a new JDK
+##################
+# Find url on https://jdk.java.net/17/ or 
+# cd downloads
+# wget https://download.java.net/java/GA/jdk17/0d483333a00540d886896bac774ff48b/35/GPL/openjdk-17_linux-x64_bin.tar.gz
+# cd ..
+# tar -xf downloads/openjdk-17_linux-x64_bin.tar.gz -C ./jdk/
+
+# To add a new Maven version
+############################
+# Find url on https://maven.apache.org/
+# cd downloads
+# wget https://dlcdn.apache.org/maven/maven-3/3.8.2/binaries/apache-maven-3.8.2-bin.tar.gz
+# cd ..
+# tar -xf downloads/apache-maven-3.8.2-bin.tar.gz -C ./maven/
+# Adjust mavenVersion below
```

Comment:
Why are we commenting out these sections? If I wanted to use my own maven or JDK, I need to rewrite a good part of the script below anyway.

Reply:
Because if you run the script 10 times, you only need to do these steps once. This is setup.
Also, choosing the JDK versions to compare etc, is the choice of the user.

We could create a script to automate the setup steps too, but that is currently out of scope.